packages/@aws-cdk/cloudformation-diff/lib/diff/template-and-changeset-diff-merger.ts

```diff
@@ -0,0 +1,211 @@
+// The SDK is only used to reference `DescribeChangeSetOutput`, so the SDK is added as a devDependency.
+// The SDK should not make network calls here
+import type { DescribeChangeSetOutput as DescribeChangeSet, ResourceChangeDetail as RCD } from '@aws-sdk/client-cloudformation';
+import { diffResource } from '.';
+import * as types from '../diff/types';
+
+export type DescribeChangeSetOutput = DescribeChangeSet;
+export type ChangeSetResourceChangeDetail = RCD;
+
+/**
+ * The purpose of this class is to include differences from the ChangeSet to differences in the TemplateDiff.
+ */
+export class TemplateAndChangeSetDiffMerger {
+  // If we somehow cannot find the resourceType, then we'll mark it as UNKNOWN, so that can be seen in the diff.
+  static UNKNOWN_RESOURCE_TYPE = 'UNKNOWN';
```

Comment:
`private static UNKNOWN_RESOURCE_TYPE`. Also, just curious, what will an unknown resourceType look like in the diff?

Reply:
```
[~] UNKNOWN nice187 nice1878D68AE26 replace
 └─ [~] QueueName (requires replacement)
     ├─ [-] nice187ANDnewAndDifferent
     └─ [+] nice187ANDnewAndDifferentsdgf
[~] UNKNOWN nice188 nice188D555BE27 replace
 └─ [~] QueueName (requires replacement)
     ├─ [-] nice188ANDnewAndDifferent
     └─ [+] nice188ANDnewAndDifferentsdgf
```

To produce this, I actually just changed the code to always set UNKNOWN. However, this should only happen if something has gone wrong... but rather than crash the program, I think this is better. Maybe we could change it to `UNKNOWN_RESOURCE_TYPE`, though

Reply:
Here's what it looks like now: 

```
[~] UNKNOWN_RESOURCE_TYPE nice8 nice828888BA3 replace
 └─ [~] QueueName (requires replacement)
     ├─ [-] nice8ANDnewAndDifferentsdgf
     └─ [+] nice8ANDdiff
```